src/java/org/apache/cassandra/db/rows/BTreeRow.java

```diff
@@ -497,6 +497,19 @@ public int dataSize()
         return Ints.checkedCast(accumulate((cd, v) -> v + cd.dataSize(), dataSize));
     }
 
+    @Override
+    public long unsharedHeapSize()
+    {
+        long heapSize = EMPTY_SIZE
+                        + clustering.unsharedHeapSize()
+                        + primaryKeyLivenessInfo.unsharedHeapSize()
+                        + deletion.unsharedHeapSize()
```

Comment:
Aside: Part of me wonders if having something like an `Accountable` interface for `unsharedHeapSize()` would make it possible to factor out some utilities for adding up the sizes of "Accountables". No action required, but I remember this being an interesting concept when I worked on Lucene in a past life.

Reply:
^_^

Reply:
no interface exists for `unsharedHeapSizeExceptData` though!